addon/-private/system/model/internal-model.js

```diff
@@ -358,6 +392,12 @@ export default class InternalModel {
 
       this._record = this.store.modelFactoryFor(this.modelName).create(createOptions);
 
+      if (hasCreateProperties && initializedRelationships.length) {
+        for (let i = 0; i < initializedRelationships.length; i++) {
+          this._relationships.get(initializedRelationships[i]).setHasData(true);
```

Comment:
Hmm, why doesn't `setDirtyHasMany` / `setDirtyBelongsTo` mark the relationship as having data?  Seems vaguely surprising to me...

Reply:
We can move this into the original iteration (I believe), but not into the methods. `setDirtyHasMany` and `setDirtyBelongsTo` are used for update code-paths as well as create-code-paths and we only mark this as hasData for creates.